docs/core/idioms/notifications.md

```diff
@@ -0,0 +1,56 @@
+# Configuring notifications
+
+Sending notifications based on events in Prefect is a common pattern. Types of notifications generally include functionality such as posting state of runs to Slack, sending a text message on failure, etc. There are many places in Prefect to hook in notification systems and this idiom will go over three possibilities.
+
+### Downstream tasks
+
+Adding tasks to your flow that respond to upstream events is one way of sending notifications. These tasks can be built in any way you would like and the Prefect task library even ships with a few [notification tasks](/api/latest/tasks/notifications.html)! Below is an example of using the [`SlackTask`](/api/latest/tasks/notifications.html#slacktask) to send a notification to Slack.
+
+::: warning Slack Webhook URL
+This code below makes use of a [Slack Webhook URL](https://api.slack.com/messaging/webhooks) as a Prefect Secret. For more information on using Prefect Secrets visit the [concept documentation](/core/concepts/secrets.html).
+:::
+
+:::: tabs
+::: tab "Functional API"
+```python
+from prefect import task, Flow
+from prefect.tasks.notifications import SlackTask
+
+@task
+def get_value():
+    return "Test SlackTask!"
+
+slack_task = SlackTask()
+
+with Flow("slack-test") as flow:
+    value = get_value()
+    slack_task(message=value)
+```
+:::
+
+::: tab "Imperative API"
+```python
+from prefect import Task, Flow
+from prefect.tasks.notifications import SlackTask
+
+class GetValue(Task):
+    def run(self):
+        return "Test SlackTask!"
+
+flow = Flow("slack-test")
+
+value = GetValue()
+slack_task = SlackTask()
+
+slack_task.set_upstream(value, key="message", flow=flow)
+```
+:::
+::::
+
+### State handlers
+
+Another common pattern for sending notifications is through [state handlers](/core/concepts/notifications.html#state-handlers). State handlers are a way of reacting to specific state changes on a task and Prefect ships with a [few state handlers](/api/latest/utilities/notifications.html) for working with different services. Outside of those default handlers it is straightforward to make completely custom handlers! For more information on this take a look at the relevant [concept documentation](/core/concepts/notifications.html).
```

Comment:
We should at least differentiate between Flow-level state handlers and task-level state handlers here.